dali/driver/quaddali.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import struct
+from time import sleep
+import serial
+
+from dali.driver.base import DALIDriver
+from dali.driver.base import SyncDALIDriver
+from dali.driver.base import Backend
+from dali.frame import BackwardFrame
+
+class QuadDALIUSBDriver(DALIDriver):
+    """``DALIDriver`` implementation for Quad-DALI-USB-Interface.
+
+    This code borrows from the HassebDALIUSBDriver.
+    """
+
+    def construct(self, command):
+        data = None
+
+        if len(command.frame) == 16:
+            address, command = command.frame.as_byte_sequence
+            data = struct.pack('BB', address, command)
+        elif len(command.frame) == 24:
+            raise ValueError('24 Bit frames are not supported (yet)')
+        else:
+            raise ValueError(f'unknown frame length: {len(command.frame)}')
+
+        return data
+
+    def extract(self, data):
+        if len(data) == 1:
+            return BackwardFrame(data[0])
+        return None
+
+class SerialBackend(Backend):
```

Comment:
The SerialBackend belongs to dali.driver.base.

Reply:
As in moving it to dali.driver.base? Just so I understand your suggestion as it was implied.

I'm unhappy that the driver ended up in this PR as it's a separate feature. I will try to untangle this mess and replace the PR.

Reply:
A git-rebase and force push should do the job :-)

Reply:
> As in moving it to dali.driver.base

exactly

Reply:
That did *not* go as planned 😭
In the end I found out how to rewrite the history of the branch, but I will have to open a new PR for it. Sorry for that 🙇

Reply:
no problem. thank you

Reply:
You shouldn't need a new pull request, but you will need to rebase your branch (probably with git rebase --interactive) and force-push.

Reply:
It took me a while to figure out the exact arguments for git rebase. I messed up my fork doing that and in a flight of panic I just deleted it. That left this PR dangling and I have yet to find a way to fix that.

Reply:
Oops!  I didn't realise that pull requests were referenced by something other than name.  I agree you'll need to close this one and start a new one.

Have you given any thought to making the memory code testable?  I usually want tests for everything in the library outside the `drivers` package.

Reply:
I have not thought about testing, yet.

I like your idea of sequences and will change the code to retrieve values to use them instead of the driver specific implementation.

Reply:
I'm just in the middle of writing a fake control gear implementation for testing sequences.  If you change your code to make use of them, you should be able to extend my fake control gear code to support memory banks for testing as well.

I hope to add this code to the Sequences pull request this afternoon; it's mostly working already, I just need to write some more comprehensive test cases.

Reply:
I've merged the simplified Sequences code into master now, including the mock DALI gear code.  If you rebase to that you could add memory bank support to the mock gear for testing.